### PR TITLE
Implement Identity Column

### PR DESCRIPTION
**Pull request by sambu16**
* Add code to implement GENERATED BY DEFAULT AS IDENTITY
 * Adjust tests

### CODE REVIEW

**Comment by patbuxton** [comment]
Hi Christoper, 
Thanks for taking the time to have a look at this. I'm trying to port over a system that already uses SQLAlchemy against Postgresql/Greenplum to be able to use HANA. The implementation on the Postgresql side makes use of many SERIAL columns so it has been that behaviour that I have been trying to replicate using the HANA dialect. 
I tried to follow the pattern on the SQLAlchemy side that was taken for the SERIAL column for the Postgresql dialect, you'll see many similarities between this change here and https://github.com/zzzeek/sqlalchemy/blob/513a6e5bcea5dccabaf86166906233542a4669e7/lib/sqlalchemy/dialects/postgresql/base.py#L2104

However, I did feel uneasy about the pre-execution of the sequence when I was porting the code over. It felt a little strange to execute a sequence to provide a value that would be generated if you didn't provide it. Also, the SQL to perform the lookup for the SYS generated sequence is pretty horrific and, like you said, subject to change - even if that might be fairly unlikely.

I honestly hadn't spotted the get_lastrowid (I'm new to Python/SQLAlchemy) - I think that would make for a much more sensible implementation - I'll see if I can get it working in that way over the next day or so.

**Comment by patbuxton** [comment]
Also, I notice that a couple of tests fail, have you seen these fail yourself?

```
FAIL test/test_suite.py::ExceptionTest_hana+hdbcli_2_0_23_0_1513691289::()::test_integrity_error
FAIL test/test_suite.py::NumericTest_hana+hdbcli_2_0_23_0_1513691289::()::test_numeric_no_decimal
```

File: setup.cfg [hunk 1]
```diff
@@ -12,3 +12,9 @@ cover-erase = 1
 [sqla_testing]
 requirement_cls=sqlalchemy_hana.requirements:Requirements
 profile_file=.profiles.txt
+
+[db]
+default=hana://UID:PWD@localhost:39044/test
```
[the remark this answers is not in the document]
Comment (patbuxton) [reply setup.cfg]: Yes, I'll remove it

File: setup.py [hunk 2]
```diff
@@ -40,7 +43,9 @@ def get_version():
         "pytest>=2.5.2",
         "mock>=1.0.1"
     ],
-    classifiers=[ # cf. http://pypi.python.org/pypi?%3Aaction=list_classifiers
+    setup_requires=['pytest-runner'],
```
[the remark this answers is not in the document]
Comment (patbuxton) [reply setup.py]: No problem :)

**Comment by SachdevaS** [comment]
@patbuxton - These are known bugs in the python driver and in HANA itself.

**Comment by kasium** [comment]
See https://github.com/SAP/sqlalchemy-hana/pull/52

**Pull request closed without merging**